dd-sdk-android/generate_core_models.gradle.kts

```diff
@@ -0,0 +1,12 @@
+import com.datadog.gradle.plugin.apisurface.ApiSurfacePlugin
+
+tasks.register(
+    "generateCoreModelsFromJson",
+    com.datadog.gradle.plugin.jsonschema.GenerateJsonSchemaTask::class.java
+) {
+    inputDirPath = "src/main/json/core"
+    targetPackageName = "com.datadog.android.core.model"
+
+    tasks.findByName(ApiSurfacePlugin.TASK_GEN_API_SURFACE)
```

Comment:
shouldn't we add a dependency on `preBuild` task? same for other new `kts` file.

Reply:
Actually we'll add a dependency on the KotlinCompile tasks

Reply:
makes sense, it is even better